src/blueapi/core/context.py

```diff
@@ -201,9 +201,18 @@ def valid(cls, value):
                 def __modify_schema__(
                     cls, field_schema: dict[str, Any], field: ModelField | None
                 ):
+                    target_type = f"{target.__module__}.{target.__qualname__}"
                     if field:
                         field_schema.update(
-                            {"type": f"{target.__module__}.{target.__qualname__}"}
+                            {
+                                "type": "string",
+                                "enum": [
+                                    k
+                                    for k, v in self.devices.items()
```

Comment:
I don't believe there is any guaranteed order for registration of plans and devices, meaning that some devices may not be registered yet.

Reply:
Isn't this method only called when the schema is called for? Does anything query schemas before everything is loaded?

Reply:
No, the schemas are generated upfront when the plan is registered with the context. I don't mind if you make them lazy instead though.